lib/xs_transport_lwt_unix_client.ml

```diff
@@ -17,19 +17,38 @@
 open Lwt
 
 (* Individual connections *)
-type channel = Lwt_unix.file_descr * Lwt_unix.sockaddr
+type channel = Lwt_unix.file_descr
+
 let create () =
-  let sockaddr = Lwt_unix.ADDR_UNIX(!Xs_transport.xenstored_socket) in
-  let fd = Lwt_unix.socket Lwt_unix.PF_UNIX Lwt_unix.SOCK_STREAM 0 in
-  lwt () = Lwt_unix.connect fd sockaddr in
-  return (fd, sockaddr)
-let destroy (fd, _) = Lwt_unix.close fd
-let read (fd, _) = Lwt_unix.read fd
-let write (fd, _) bufs ofs len =
-	lwt n = Lwt_unix.write fd bufs ofs len in
-	if n <> len then begin
-		fail End_of_file
-	end else return ()
+  ( match Xs_transport.choose_xenstore_path () with
+    | None ->
+      Printf.fprintf stderr "Failed to find xenstore socket. I tried the following:\n";
+      List.iter (fun x -> Printf.fprintf stderr "  %s\n" x) (Xs_transport.get_xenstore_paths ());
```

Comment:
I assume it's pretty unlikely that the `xenstored_socket` reference is changed since the check in `match Xs_transport.choose_xenstore_path ()`, so that the paths it prints out are different from the ones that it actually checked.